orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py

```diff
@@ -61,19 +61,25 @@ def nll_loss(g, self, target, weight, reduction, ignore_index):
     output.setType(self.type())
     return output
 
-
 @register_symbolic('embedding')
 def embedding(g, weight, indices, padding_idx, scale_grad_by_freq, sparse):
-    output = g.op("com.microsoft::ATenOp", weight, indices, padding_idx, scale_grad_by_freq, sparse,
-                  name_s='aten::embedding')
+    output = None
+    padding_idx_int, scale_grad_by_freq_bool, sparse_bool = sym_help._maybe_get_const(padding_idx, 'i'), \
+        sym_help._maybe_get_const(scale_grad_by_freq, 'b'), sym_help._maybe_get_const(sparse, 'b')
+    # negative padding_idx_int implies the original padding_idx was None
+    if scale_grad_by_freq_bool or sparse_bool or (padding_idx_int is not None and padding_idx_int >= 0):
+        output = g.op("com.microsoft::ATenOp", weight, indices, padding_idx, scale_grad_by_freq, sparse,
+                    name_s='aten::embedding')
+    else:
+        output = g.op("Gather", weight, indices, outputs=1)
```

Comment:
Can be directly exported with GatherInternal?

Reply:
Replacement of `Gather` to `GatherInternal` should only happen for cuda/rocm execution providers. For other providers, we should probably execute the `Gather` node. So, exporting with `GatherInternal` would mean we would need to make `GatherInternal` available for other execution providers as well.